### PR TITLE
WIP: feat: Add a bunch of mips compilers

### PR DESCRIPTION
**Pull request by maringuu**
Companion to https://github.com/compiler-explorer/gcc-cross-builder/pull/65

There are still a lot of compilers in the linked PR that are not covered here,  so this is a draft.

### CODE REVIEW

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -2243,13 +2243,18 @@ compiler.mipsg1320.demangler=/opt/compiler-explorer/mips/gcc-13.2.0/mips-unknown
 
 ## MIPS 64
 group.mips64.groupName=MIPS64 GCC
-group.mips64.compilers=mips64g1210:mips64g1220:mips64g1230:mips64g1310:mips64g1320:mips564:mips112064
+group.mips64.compilers=mips64g1210:mips64g1220:mips64g1230:mips64g1310:mips64g1320:mips564:mips112064:mips64g1050
```
Comment (dkm) [inline etc/config/c++.amazon.properties:2246]: small nit: can you keep the (kind) of order we have here? Here, the list is ordered by size (32/64) and then by versions.
That's a very small change, but when we introduce a new GCC version on the 20+ targets we have, it's harder if every target has its own ordering/naming.

**Comment by dkm** [comment]
In the gcc-cross-builder, there's a script that you can use to update all the config at once: GCC knows about C, C++, but that's not the end of the list... go, Ada, fortran, objc, objc++, modula2, rust, ...

**Comment by dkm** [comment]
Don't hesitate to post any progress or ask for guidance, we'll do our best to help :)

**Comment by maringuu** [comment]
> Don't hesitate to post any progress or ask for guidance, we'll do our best to help :)

Thank you! I didn't have time yet to come back to you, but I already read and appreciated your comments. So thank you!
I'll probably continue working on this on Thursday :)

**Comment by dkm** [comment]
I've started the 24 builds (can be tracked here: https://github.com/compiler-explorer/infra/actions job 550->573)

**Comment by dkm** [comment]
The compilers are built, but we need infra changes to install them.

**Comment by maringuu** [comment]
Yeah, I haven't had the time yet to add the compilers.
I will add them eventually, but feel free do do it yourself in the meantime.

**Comment by maringuu** [comment]
I will not have time for this until march. Sorry. But I still have this in mind, as one of my projects depends on these compilers being build.

**Comment by dkm** [comment]
No worries. Most of the job is done, so I may push it over the finish line later :) thanks again

**Comment by dkm** [comment]
> No worries. Most of the job is done, so I may push it over the finish line later :) thanks again
> […](#)
> On January 13, 2024 4:09:41 PM UTC, Marten Ringwelski ***@***.***> wrote: I will not have time for this until march. Sorry. But I still have this in mind, as one of my projects depends on these compilers being build. -- Reply to this email directly or view it on GitHub: [#5723 (comment)](https://github.com/compiler-explorer/compiler-explorer/pull/5723#issuecomment-1890554344) You are receiving this because you were assigned. Message ID: ***@***.***>

I've done some work on that. 4.9.4, 5.5.0 and 9.5.0 should be done. I see your current PR mentions 10.5.0, but I haven't seen this config submitted in the gcc-cross-builder repository... Don't hesitate to contribute it, when you want. But ATM, I think we're done and don't have anything "pending". Please correct me if I'm wrong.

**Comment by maringuu** [comment]
Nice work, thank you!

>  I see your current PR mentions 10.5.0

I think I confused 9.5.0 as 10.5.0.

Closing as all compilers are added.

**Pull request closed without merging**